x/evm/keeper/state_transition.go

```diff
@@ -325,13 +297,15 @@ func (k *Keeper) RefundGas(msg core.Message, leftoverGas uint64) error {
 		// no refund, consume gas and update the tx gas meter
 	}
 
+	return nil
+}
+
+func (k *Keeper) setGasMeterAndConsumeGas(msg core.Message, gasUsed uint64) {
```

Comment:
This part was extracted from `RefundGas` in the original code, not sure it is enough